### PR TITLE
Fix/update time timezone

### LINKED ISSUE #135: get_last_updated_time() doesn’t contain timezone
**Issue Overview**
Currently, `get_last_updated_time()` doesn’t contain a timezone. It’s always in format `YYYY-mm-dd H:i:s`. However, if you try to compare it with the current installed translations (e. g. as you do in (traduttore-registry)[https://github.com/wearerequired/traduttore-registry/blob/master/inc/namespace.php#L68]) this test could result in an unexpected result.

I updated a translation in `2018-12-03 11:29:25` local time, which is GMT+1. After updating the translations in my WordPress, their `PO-Revision-Date` ffrom `wp_get_installed_translations()` is correctly set to `2018-12-03 10:29:25+0000`. However, the translations API does print the following `updated` date: `2018-12-03 10:29:25`. The result was that the update notification for outstanding translations never gets removed.

I double-checked the compare by using the following code:

```php
<?php
$with_timezone = new Datetime( '2018-12-03 10:29:25+0000' );
$without_timezone = new Datetime( '2018-12-03 10:29:25' );

switch ( true ) {
    case $with_timezone > $without_timezone:
        echo 'With > Without';
        break;
    case $with_timezone === $without_timezone:
        echo 'With = Without';
        break;
    case $with_timezone < $without_timezone:
        echo 'With < Without';
        break;
}
```

Result:
```
With < Without
```

**Versions**

* Browser: Safari
* PHP: 7.1.24
* WP-CLI: 2.0.1
* GlotPress: 2.3.1
* WordPress: 4.9.8

### PR DESCRIPTION
**Pull request by swissspidy**
**Description**
Fixes #135 by using `DATE_FORMAT` to save the last updated time with timezone information. Also makes sure the getters always return DateTime objects, which makes handling this data easier anyway.

**How has this been tested?**
Updated unit tests accordingly.

**Types of changes**
Bug fix (non-breaking change which fixes an issue)

**Checklist:**
- [x] My code is tested.
- [x] My code follows the WordPress code style. 
- [x] My code has proper inline documentation.

### CODE REVIEW

File: inc/ZipProvider.php [hunk 1]
```diff
@@ -213,12 +214,12 @@ protected function get_zip_filename() : string {
 	 *
 	 * @since 2.0.0
 	 *
-	 * @return string Last build time.
+	 * @return DateTime Last build time.
 	 */
-	public function get_last_build_time() :? string {
+	public function get_last_build_time() :? DateTime {
 		$meta = gp_get_meta( 'translation_set', $this->translation_set->id, static::BUILD_TIME_KEY );
```
Comment (ocean90) [inline inc/ZipProvider.php]: Should `generate_zip_file()` also be updated to store in `DATE_ATOM` format?

File: inc/ZipProvider.php [hunk 2]
```diff
@@ -213,12 +214,12 @@ protected function get_zip_filename() : string {
...
 
-		return $meta ?: null;
+		return $meta ? new DateTime( $meta ) : null;
```
Comment (ocean90) [inline inc/ZipProvider.php]: Timezone needed?

File: inc/Project.php [hunk 3]
```diff
@@ -421,12 +422,12 @@ public function set_text_domain( string $name ): bool {
...
 	 *
-	 * @return null|string Last updated time if stored, null otherwise.
+	 * @return null|DateTime Last updated time if stored, null otherwise.
 	 */
-	public function get_last_updated_time(): ?string {
+	public function get_last_updated_time(): ?DateTime {
 		$time = gp_get_meta( 'project', $this->project->id, static::UPDATE_TIME_KEY );
 
-		return $time ?: null;
+		return $time ? new DateTime( $time ) : null;
```
Comment (ocean90) [inline inc/Project.php]: Does this need to set the `DateTimeZone` to UTC by default? Probably for existing metas.

File: inc/Project.php [hunk 4]
```diff
@@ -438,7 +439,7 @@ public function get_last_updated_time(): ?string {
 	 * @return bool Whether the data was successfully saved or not.
 	 */
 	public function set_last_updated_time( string $time ): bool {
```
[the remark this answers is not in the document]
Comment (swissspidy) [reply inc/Project.php]: Hesitated a bit to do that, but yeah it makes sense. Also solves most of the other issues.

**Pull request merged**